src/components/authentication/if-authenticated.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import { AuthenticationContext } from '../../context/authentication';
+
+export interface Properties {
+  show?: boolean;
+  hide?: boolean;
+}
+
+export class IfAuthenticated extends React.Component<Properties> {
+  static contextType = AuthenticationContext;
```

Comment:
adding context in this way makes the structure of the context less discoverable, and it also makes it harder to test. instead, i would add an HOC so that the context is just injected through props. 

something like this: https://github.com/zer0-os/zOS-component-library/blob/main/src/components/app-layout/context.tsx#L17

and consumed like this: https://github.com/zer0-os/zOS-component-library/blob/main/src/components/app-layout/app-layout.tsx#L44